scripts/publish.sh

```diff
@@ -31,32 +31,34 @@ package_json() {
 # check preconditions
 # make sure we can read the package.json
 [[ -f package.json ]] || error "could not locate package.json in directory $(pwd). Publish must be run from the package root."
-git rev-parse --abbrev-ref "${1:-HEAD}" >/dev/null 2>&1 || error "branch $1 does not exist"
-PACKAGE_NAME="$(package_json 'name')"
-PACKAGE_VERSION="$(package_json 'version')"
-BRANCH_NAME="$(git rev-parse --abbrev-ref "${1:-HEAD}")"
+git rev-parse --verify "${1:-HEAD}" >/dev/null 2>&1 || error "ref $1 does not exist"
+REF_NAME="$(git rev-parse --abbrev-ref "${1:-HEAD}")"
 
 # warn if release is not rel/something
-[[ "$BRANCH_NAME" == "rel/*" ]] || \
-confirm "Branch $BRANCH_NAME does not look like a release branch. Are you sure you want to publish this branch?"
+[[ "$REF_NAME" =~ rel/.* || "$REF_NAME" =~ [0-9]+\.[0-9]+\.[0-9]+ ]] || \
+confirm "Ref $REF_NAME does not look like a release branch or version tag. Are you sure you want to publish this ref?"
```

Comment:
Why not allow continuing forward? What if we for some reason want to release a branch that doesn't match `rel/*` or `*.*.*`?

Reply:
> What if we for some reason want to release a branch that doesn't match rel/* or *.*.*?

Why would you want to allow that? Basically, you'd allow to publish anything from any random branch or tag. Sounds like a bad and unacceptable process to me.

Reply:
> FYI, unspents packages have been published from origin/master (based on release tags) instead of rel/*.

That's why I want to allow it. It's up to the human running the script to make the correct decisions about what ref they want to release. They could always side step the publish script and do it manually, but we don't want to encourage that so the publish script should be flexible and provide warnings when things may not be correct, but allow the human to override if necessary.

Reply:
You see it from your Developer's point of view, and I see it from the release point of view which do not seem to converge or agree. Leaving alone the SOC2 requirements.

Even with the new bitgo-microservices, although the entire CI/CD pipeline is fully automated, it only allows certain release tag being signed by certain authorized BRE individual before auto deploying to production. Nobody else can.